src/Storages/StorageURL.cpp

```diff
@@ -217,6 +219,8 @@ namespace
                     uri_options.size() == 1,
                     download_threads);
 
+                total_size += getFileSizeFromReadBuffer(*read_buf);
```

Comment:
there can be a problem if multiple URLs are used because of the lazy initialization of each of them.
Maybe it's okay to fetch sizes for each URL first. 🤔

Reply:
Can you just use list of urls that are left x some average that you have? If you read 1 file out of 1000 it is ok to show 0.1% progress ignoring their actual size.

Reply:
> Maybe it's okay to fetch sizes for each URL first. 

No, not ok.
We can have tens of millions of URLs easily.

It's ok if the progress bar jumps backward as new info becomes available.

Reply:
> tens of millions of URLs

Yes, not a good option then.